### ISSUE TITLE
Mismatching react version with react 17

### ISSUE THREAD

**Issue by ozanmanav** [formium/formik#3167]
Hello. When will the master branch of the formic repo be released? Now, the old react version in version 2.2.6 causes problems with monorepo.

When I check the 2.2.6 package.json it seems used react 16.x version. But we're using react 17.0.1 with yarn workspace. Because of this, it is crashing. I have to use 17 for the formic. I'm seeing the master branch already upgraded.

 How can we solve this problem?

![image](https://user-images.githubusercontent.com/19241756/116487852-9ea4df80-a899-11eb-931b-13c8beaf0d09.png)

**Comment by ozanmanav** [comment]
I guess it is not about the react version. It is creating another react folder with old version in the formik folder like this:

![image](https://user-images.githubusercontent.com/19241756/116521019-1eec3480-a8dc-11eb-86a2-cf59f53bad4b.png)
![image](https://user-images.githubusercontent.com/19241756/116521050-27dd0600-a8dc-11eb-948e-1665374ba575.png)

**Comment by johnrom** [comment]
You can add a property to your root workspace package.json:

```json
"resolutions": {
  "react": "^17",
  "react-dom": "^17"
}
```

**Comment by jaredpalmer** [comment]
@johnrom is this is as easy as just cutting a new release?

**Comment by johnrom** [comment]
Hmm.. not really. I don't really know why the user is receiving this issue -- Yarn doesn't pull dev dependencies into the users node modules.

@ozanmanav Are you using `link` to connect a physically downloaded / cloned version of Formik to your workspace? If you are simply using Formik as a dependency via `package.json:dependencies`, it will not (or shouldn't) download React for you at all, as it is a peer dependency.

If you are using "include peer dependencies" or something in your workspace, Idk I've never used it, but I would instead install peer dependencies manually.

If we want to update peer dependencies (we should), the format is a little different. I messed around with it in one of my old PRs but it isn't any of the ones I currently have open and I couldn't find it. The format is something like this:

```json
"peerDependencies": {
  "react": ">=17 || >=16.8.0"
}
```

I just wrote that out by hand so it might not be exactly that. YMMV.
